packages/workbox-core/utils/indexedDBHelper.mjs

```diff
@@ -0,0 +1,122 @@
+/**
+ * This is a wrapper that makes it easier to use IDB.
+ */
+class DBWrapper {
+  /**
+   * Wraps a provided Database.
+   * @param {IndexedDB} idb
+   * @param {string} storename
+   */
+  constructor(idb, storename) {
+    this._db = idb;
+    this._storename = storename;
+  }
+
+  /**
+   * Get a value for a given ID.
+   * @param {Object} key
+   * @return {Promise<string>}
+   */
+  get(key) {
+    return new Promise((resolve, reject) => {
+      const transaction = this._db.transaction(this._storename, 'readonly');
+
+      transaction.onerror = () => {
+        // Don't forget to handle errors!
+        reject(transaction.error);
+      };
+
+      const getRequest = transaction.objectStore(this._storename).get(key);
+      getRequest.onsuccess = () => resolve(getRequest.result);
+    });
+  }
+
+  /**
+   * Put a value in the database for a given id.
+   * @param {Object} key
+   * @param {Object} value
+   * @return {Promise}
+   */
+  put(key, value) {
+    return new Promise((resolve, reject) => {
+      const transaction = this._db.transaction(this._storename, 'readwrite');
+
+      transaction.onerror = () => {
+        // Don't forget to handle errors!
+        reject(transaction.error);
+      };
+
+      const request = transaction.objectStore(this._storename).put(value, key);
+      request.onsuccess = () => resolve();
+    });
+  }
+
+  /**
+   * Delete a value in the database with a given id.
+   * @param {Object} key
+   * @return {Promise}
+   */
+  delete(key) {
+    return new Promise((resolve, reject) => {
+      const transaction = this._db.transaction(this._storename, 'readwrite');
+
+      transaction.onerror = () => {
+        // Don't forget to handle errors!
+        reject(transaction.error);
+      };
+
+      const request = transaction.objectStore(this._storename).delete(key);
+      request.onsuccess = () => resolve();
+    });
+  }
+}
+
+/**
+ * This class will be used to create a generic IndexedDB class for Workbox
+ * to use, removing the annoying parts of the API.
+ */
+class IndexedDBHelper {
+  /**
+   * You should never construct this directly.
+   */
+  constructor() {
+    this._opendedDBs = {};
+  }
+
+  /**
+   * Get an opened IndexedDB.
+   * @param {string} name
+   * @param {Number} version
+   * @param {string} storename
+   * @param {Object} objectStoreOptions
+   * @return {Promise<IDBObjectStore>}
+   */
+  async getDB(name, version, storename, objectStoreOptions) {
+    if (this._opendedDBs[name]) {
+      return this._opendedDBs[name];
```

Comment:
Some issues here:

This method accepts lots of params, but if I call it twice with the same `name` value and different `version`, `storename`, or `objectStoreOptions` values, I'll just get the first `DBWrapper` instance back since the `names` are the same.

Also, I think you may want to remove the `version` param (or if you keep it, allow consumers to specificy the `onupgradeneeded` callback).

Reply:
The version and storename should not be dynamically changed, if it is, it'll be a massive world of pain. Do you want me to code to support this?